Content.Server/GameObjects/Components/Fluids/PuddleComponent.cs

```diff
@@ -0,0 +1,356 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Content.Server.GameObjects.Components.Chemistry;
+using Content.Shared.Chemistry;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using Timer = Robust.Shared.Timers.Timer;
+
+namespace Content.Server.GameObjects.Components.Fluids
+{
+    /// <summary>
+    /// Puddle on a floor
+    /// </summary>
+    [RegisterComponent]
+    public class PuddleComponent : Component
+    {
+        // Current design: Something calls the SpillHelper.Spill, that will either
+        // A) Add to an existing puddle at the location (normalised to tile-center) or
+        // B) add a new one
+        // From this every time a puddle is spilt on it will try and overflow to its neighbours if possible,
+        // and also update its appearance based on volume level (opacity) and chemistry color
+        // Small puddles will evaporate after a set delay
+
+        // TODO: 'leaves fluidtracks', probably in a separate component for stuff like gibb chunks?;
+        // TODO: Add stuff like slipping -> probably in a separate component (for stuff like bananas) and using BumpEntMsg
+
+#pragma warning disable 649
+        [Dependency] private readonly IEntityManager _entityManager;
+        [Dependency] private readonly IEntitySystemManager _entitySystemManager;
+        [Dependency] private readonly IMapManager _mapManager;
+#pragma warning restore 649
+
+        // based on behaviour (e.g. someone being punched vs slashed with a sword would have different blood sprite)
+        // to check for low volumes for evaporation or whatever
+
+        // TODO: If the underlying tile becomes space then this should be deleted
```

Comment:
You can hook `IMapManager.TileChanged` with an entity system and delete any puddles on the tile if it gets set to a tile like space.